feat#5 create comment interface

PeindreLaColline opened the pull request:
# PR Description
> comment interface 작성

## Related Issue

close #5

## Help
> 인터페이스 이정도면 될까요?

PeindreLaColline committed:
feat#5 create comment interface

PeindreLaColline committed:
refact#5 CommentService interface

commentService 인터페이스 반환값 및 메서드 명 변경

PeindreLaColline committed:
feat#5 create CommentDtoRes/Req

commentDtoRes 및 Req 추가

weejinyoung commented on src/main/java/mini/backend/comment/CommentDtoReq.java:
```diff
@@ -0,0 +1,17 @@
+
+@Getter
+@AllArgsConstructor
+public class CommentDtoReq {
+    private final Long commentId;
```
1. 이 Dto 는 어떤 요청에 쓰이는 Dto 인가요? 모든 Req 에 쓰이는 Dto 인가요?
2. 만약 create 메소드에도 이 Dto 가 쓰인다면 commentId 를 클라이언트가 보낼 수 있을까요? commentId 는 데이터베이스가 생성해주는 것으로 알고있습니다만!

PeindreLaColline replied:
어디쓰이는지 고려해서 req res 둘 다 수정했습니당

weejinyoung commented on src/main/java/mini/backend/comment/CommentService.java:
```diff
@@ -0,0 +1,20 @@
+import java.util.List;
+
+public interface CommentService {
+    //POST
+    void create(Comment comment);
```
Comment 클래스를 인자로 받는 것으로 하셨는데,
처음 인터페이스 설계를 할 때 CommentService 는 컨트롤러 단에서 참조하는 것으로 약속한 기억이 있습니다
그렇다는 것은 Comment 클래스의 인스턴스를 컨트롤러 단에서 만들어서 인자로 보내줘야 한다는 것인데 맞을까요?

PeindreLaColline replied:
CommentDtoReq로 수정했읍니다

weejinyoung commented on src/main/java/mini/backend/comment/CommentDtoRes.java:
```diff
@@ -0,0 +1,19 @@
+@AllArgsConstructor
+public class CommentDtoRes {
+    private final Long commentId;
+    private final User userId;
+    private final Post postId;
```
변수명은 userId 인데 타입은 User 네요! 의도하신 바가 맞으실까요?

PeindreLaColline replied:
그 부분 사죄하고 반성합니다 수정했습니다 커밋은 나중에 한꺼번에 할게요

prin81208 commented:
확인했습니다 고생하셨습니다 !!

PeindreLaColline committed:
refact#5 CommentDto/Req, Comment Service

피드백 사항 반영해서 수정했습니다
CommentDtoReq에 주석 참고해주시면 감사하겠습니다

PeindreLaColline committed:
Merge remote-tracking branch 'origin/main' into feat#5-create-comment-interface

PeindreLaColline committed:
refact#5 modify commentDtoReq/Res, commentService

Req: commentId는 URL로 들어오니 제외함.
Res: UserDtoRes 추가.

PeindreLaColline committed:
Merge remote-tracking branch 'origin/main' into feat#5-create-comment-interface

# Conflicts:
#	src/main/java/mini/backend/user/UserDtoRes.java

PeindreLaColline committed:
refact#5 comment 패키지 피드백 반영, post 패키지 pull, entity 클래스 변수명 변경

1. 제가 맡았던 comment 패키지 관련 클래스들 피드백 반영해서 수정했습니다.
2. 혜원님이 맡았던 post 패키지 풀 받아서 제 플젝에 머지했습니다
3. entity 클래스 변수명 관련 피드백 받았던 부분 수정했습니다 (ex: userId -> user)
4. 위의 세 가지 다 반영해서 한꺼번에 커밋합니다
